test/test-filter-stream.sh

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env bash
+
+. "test/testlib.sh"
+
+if [ "$GIT_LFS_USE_LEGACY_FILTER" == "1" ]; then
+  echo "skip: $0 (filter stream disabled)"
+  exit
+fi
+
+ensure_git_version_isnt $VERSION_LOWER "2.10.0"
+
+begin_test "filter stream: checking out a branch"
+(
+  set -e
+
+  reponame="filter_stream_checkout"
```

Comment:
I call it "filter process" in the Git documentation. "Filter stream" was the initial name too many months ago 😄